File: miscellaneous/sorting/insertion-sort/20221120/insertion-sort.php
<?php

declare(strict_types=1);

function insertionSort(array $nums): array
{
    if (count($nums) < 2) {
        return $nums;
    }

    for ($i = 1; $i < count($nums); ++$i) {
        $current = $nums[$i];

        for ($j = $i - 1; $j >= 0 && $current < $nums[$j]; --$j) {
            $nums[$j+1] = $nums[$j];
        }

        $nums[++$j] = $current;
    }

    return $nums;
}

File: miscellaneous/sorting/insertion-sort/20221120/insertion_sort.py
from typing import List


def insertion_sort(nums: List[int]) -> List[int]:
    if len(nums) < 2:
        return nums

    for i in range(1, len(nums)):
        current = nums[i]
        j = i - 1

        while j >= 0 and current < nums[j]:
            nums[j+1] = nums[j]
            j -= 1

        j += 1
        nums[j] = current

    return nums
